doc/Migration_guide_0.4.md

```diff
@@ -0,0 +1,84 @@
+# Migrating Java service to v0.4
+
+## Update the client apps
+Update the clients to submit transactions in the new format, introduced in Exonum 0.10.
+The most notable change in the format is that the public key of the transaction author 
+must always be included in the transaction message. This allows the framework to verify 
+the message signature with no conversion to an executable transaction.
+
+If the client application is in Java, use `com.exonum.binding.common.message.TransactionMessage`;
+if it is in Java Script, use the latest version of the JS light client.
+
+The light client shall submit transactions directly to the core instead of a custom endpoint
+in the service controller. <!-- TODO: which core endpoint? -->
+
+## Update the service
+### Use the new message format
+
+EJB 0.4 accepts the transactions via a system endpoint and passes the payload to the `Service`
+implementation to convert into an executable transaction. The `TransactionConverter` now accepts
+a `RawTransaction` that includes only the transaction identifiers (service id and transaction id)
+and its payload — the serialized transaction parameters (equivalent to `Message#getBody` from v0.3).
+
+Also, consider removing the public key of the transaction author from its _payload_ because
+it is now included in a standard field of the message, and is accessible in `Transaction#execute`
+from the passed context: `TransactionContext#getFooKey`.
+
+### Update `Transaction`s
+The following methods are **removed** from `Transaction` interface:
+  - `isValid`. In v0.4 the framework verifies the cryptographic signature itself, using the
+  public key from the message. If your service performs any additional checks, 
+  move them to the constructor of your transaction or to `TransactionConverter`, whichever 
+  works best.
+  - `info`. It is no longer possible to serialize the transaction in JSON in core endpoints.
+  <!-- ^ Or shall we keep it? -->
```

Comment:
Now I think that we actually need this method. Core uses `serde::Serialize` implementation to represent `Transaction` in JSON, but we cannot directly implement `Serialize` on our `TransactionProxy` - so we need some helper method that will return optional JSON description

Reply:
I think we need to be able to serialize our transactions in JSON.

Reply:
Discussed today that we will keep `info` as is, but also consider pulling it into the framework code (e.g., by using a reflective serializer such as Gson to convert a tx instance to JSON). As it won't _always_ work as one might expect, we must also design a way to extend/override this mechanism.